src/util/DataResolver.js

```diff
@@ -78,11 +79,16 @@ class DataResolver extends null {
 
   /**
    * Resolves a Base64Resolvable to a Base 64 image.
-   * @param {Base64Resolvable} data The base 64 resolvable you want to resolve
-   * @returns {?string}
+   * @param {Base64Resolvable|Blob} data The base 64 resolvable you want to resolve
+   * @returns {Promise<string>}
    */
-  static resolveBase64(data) {
+  static async resolveBase64(data) {
     if (Buffer.isBuffer(data)) return `data:image/jpg;base64,${data.toString('base64')}`;
+    if (data instanceof Blob) {
```

Comment:
Question: don't blobs have a type too? If they do, can't we use that for the `data:content-type`?

Reply:
Yes, fixed in fd18ecf64d2ae364657159eb6128ce235c9c13cc. If the type isn't specified, it's an empty string so I still default it to `image/jpg` though.